src/Common/src/Interop/Windows/mincore/Interop.GetNamedPipeInfo.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Kernel32, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool GetNamedPipeInfo(
```

Comment:
Looked at it closer and our only option is to use out int for all. However as mentioned earlier changing the signature here will have performance implications. The only way to achieve it would be to un-marshal all the out_opt parameters and then use only one required. For now, I will chose multiple call-sites over marshaling overhead. Let me know if you disagree.